internal/jptrace/translator_test.go

```diff
@@ -88,28 +88,28 @@ func TestProtoToTraces_AddsWarnings(t *testing.T) {
 
 	assert.Equal(t, 2, traces.ResourceSpans().Len())
 
-	rs1 := traces.ResourceSpans().At(0)
-	assert.Equal(t, 1, rs1.ScopeSpans().Len())
-	ss1 := rs1.ScopeSpans().At(0)
-	assert.Equal(t, 2, ss1.Spans().Len())
+	spanMap := make(map[string]ptrace.Span)
+
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
```

Comment:
considering how often we seem to be doing this I would create a helper 
```go
package jptrace

func SpanMap[K](tr ptrace.Traces, keyFn func (ptrace.Span) K) map[K]ptrace.Span
```

Reply:
sounds good - can do in a follow-up PR